include/gauge_fix_ovr_hit_devf.cuh

```diff
@@ -264,82 +245,82 @@ namespace quda {
       //Get the two indices for the SU(N) matrix
       IndexBlock<NCOLORS>(block, p, q);
 
-      if ( threadIdx.x < blockSize ) {
+      if (mu == 0) {
         elems[tid] = link(p,p).x + link(q,q).x;
         elems[tid + blockSize] = -(link(p,q).y + link(q,p).y);
         elems[tid + blockSize * 2] = -(link(p,q).x - link(q,p).x);
         elems[tid + blockSize * 3] = -(link(p,p).y - link(q,q).y);
       }
-      __syncthreads();
-      if ( threadIdx.x < blockSize * 2 && threadIdx.x >= blockSize ) {
+      cache.sync();
+      if (mu == 1) {
         elems[tid] += link(p,p).x + link(q,q).x;
         elems[tid + blockSize] -= (link(p,q).y + link(q,p).y);
         elems[tid + blockSize * 2] -= (link(p,q).x - link(q,p).x);
         elems[tid + blockSize * 3] -= (link(p,p).y - link(q,q).y);
       }
-      __syncthreads();
-      if ( threadIdx.x < blockSize * 3 && threadIdx.x >= blockSize * 2 ) {
+      cache.sync();
+      if (mu == 2) {
         elems[tid] += link(p,p).x + link(q,q).x;
         elems[tid + blockSize] -= (link(p,q).y + link(q,p).y);
         elems[tid + blockSize * 2] -= (link(p,q).x - link(q,p).x);
         elems[tid + blockSize * 3] -= (link(p,p).y - link(q,q).y);
       }
```

Comment:
This is beyond the scope of this PR, but aren't these two code blocks exactly the same? _Should_ they be the same for different directions?

Reply:
This code block is intentional, but I have rewritten it to make it cleaner (a86aa2ce5).  Here we want to do an ordered reduction in shared memory, with different warps (each with unique `mu` values) needing to take turn to include their contribution.